pkg/server/container_create.go

```diff
@@ -768,6 +733,70 @@ func defaultRuntimeSpec() (*runtimespec.Spec, error) {
 	return spec, nil
 }
 
+// generateSeccompSpecOpts generates containerd SpecOpts for seccomp.
+func generateSeccompSpecOpts(seccompProf string, privileged, seccompEnabled bool) (containerd.SpecOpts, error) {
+	// Set seccomp profile
+	if seccompProf == runtimeDefault || seccompProf == dockerDefault {
+		// use correct default profile (Eg. if not configured otherwise, the default is docker/default)
+		seccompProf = seccompDefaultProfile
+	}
+	if !seccompEnabled {
+		if seccompProf != "" && seccompProf != unconfinedProfile {
+			return nil, fmt.Errorf("seccomp is not supported")
+		}
+		return nil, nil
+	}
+	if privileged {
```

Comment:
Maybe we could put the logic to the top of function.

If `privileged==true`, we could not care about the values of  `seccompEnabled` and `seccompProf`.

Reply:
Will do, make sense.

Reply:
Actually, CRI doesn't define the expected behavior clearly. We are not following docker's behavior.
https://github.com/moby/moby/blob/master/daemon/oci_linux.go#L751
https://github.com/moby/moby/blob/8af4db6f002ac907b6ef8610b237879dfcaa5b7a/daemon/seccomp_linux.go#L20

Reply:
Well there ya go.. in the absence of a defined spec... use the default behavior of the prior code

Reply:
> If privileged==true, we could not care about the values of seccompEnabled and seccompProf.

@miaoyq Done